week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product/Product.ts

```diff
@@ -0,0 +1,96 @@
+import {v4} from 'uuid';
+import {IProduct, IProductWithPrice, PRODUCT_TYPE} from './Product.types';
+
+class Product {
+  private readonly _id: string;
+  private _name: string;
+  private _amount: number;
+
+  public constructor({ name, amount }: IProduct) {
+    this._id = v4();
+    this._name = name;
+    this._amount = amount;
+  }
+
+  public get id(): string {
+    return this._id;
+  }
+
+  public get name(): string {
+    return this._name;
+  }
+
+  public get amount(): number {
+    return this._amount;
+  }
+
+  public set amount(value: number) {
+    if (value >= 1) {
+      this._amount = value;
+    } else {
+      throw Error('Invalid amount');
+    }
+  }
+
+  public update({ name, amount }: Partial<IProduct>): void {
```

Comment:
`:Partial<Product>` - to by wystarczyło bo interface są nadmiarowe (szczegóły poniżej)